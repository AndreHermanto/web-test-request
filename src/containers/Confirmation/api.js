/**
 * Returns the test request based on id
 * @return {Object[]} json Currently returns an array of test panel object - including id and label
 */
export function getTestRequestById(id) {
  return fetch(`${process.env.REACT_APP_BASE_URL}/test-requests/${id}`)
    .then((response) => response.json())
    .then((json) => json.data);
}
