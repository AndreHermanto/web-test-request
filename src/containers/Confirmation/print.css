.printMe {display: none;}

@media print {
  .noPrint {display: none;}
  .printMe {display: block;}
}