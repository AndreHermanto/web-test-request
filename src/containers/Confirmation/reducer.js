export function initData(data, latestRequestID, createdDateTime) {
  var state = {
    form: data,
    print: null
  }
  state.form = Object.assign({}, state.form, {
    latestRequestID: latestRequestID,
    createdDateTime: createdDateTime
  });
  return Object.assign({}, state);
}

/**
 * Set "print" state to determine which type of report to print.
 * @param {Object} state Targeted state to be changed.
 * @param {Integer} value Value to be set.
 */
export function setPrintType(state, value) {
  return Object.assign({}, state, {
    print: value
  });
}