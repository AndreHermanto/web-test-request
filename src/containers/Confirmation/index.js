import React, { Component } from 'react';
import { Row, Col } from 'react-bootstrap';
import styled from 'styled-components';
import {
  initData,
  setPrintType 
} from './reducer';
import PrintRecord from './components/PrintRecord';
import PrintBloodCollection from './components/PrintBloodCollection';
import './print.css'
import { 
  FormButton
} from './../../components/SharedStyle';

const ConfirmationBox = styled.div`
  box-shadow: 0 0 5px rgb(227, 231, 241);  
  width: 80%;
  padding: 20px;
  margin-bottom: 10px;
  margin-left: 10%;
  background-color: #fff;
`;

const ConfirmationHeading = styled.h1`
  text-align:center;
  color: rgba(51, 51, 51, 0.75);
`;
const ConfirmationNote = styled.h4`
  font-weight: 200;
  text-align:center;
  margin-top: 10px;
`;

/**
* Confirmation - UI for ordering type of tests, selecting disorder and related genes for testing.
*/
class Confirmation extends Component {
  constructor(props) {
    super(props);
    this.handlePrintRecordButtonClick = this.handlePrintRecordButtonClick.bind(this);
    this.handlePrintBloodCollectionButtonClick = this.handlePrintBloodCollectionButtonClick.bind(this);
    this.handleRedirect = this.handleRedirect.bind(this);
    this.state = initData();
  }
  
  handlePrintButtonClick(type) {
    this.setState(setPrintType(this.state, type));
    setTimeout(() => {
      window.print();
    }, 200);
  }
  
  handlePrintRecordButtonClick() {
    this.handlePrintButtonClick(1);
  }
  
  handlePrintBloodCollectionButtonClick() {
    this.handlePrintButtonClick(2);
  }

  handleRedirect(place) {
    if(place === 'home') {
      this.props.route.onClean();
      this.props.router.push('/step1'); 
    }
    else {
      window.location.replace('https://www.genome.one/'); 
    }
  }
  
  render() {
    return (
      <div>
        <ConfirmationBox className="noPrint">
            <ConfirmationHeading> Thank you!</ConfirmationHeading>
            <ConfirmationNote>
              You can now:
            </ConfirmationNote>
            <Row>
              <Col md={6}>
                <FormButton onClick={this.handlePrintRecordButtonClick} style={{marginLeft: '20%'}}>
                  Print out form for your records
                </FormButton>
              </Col>
              <Col md={6}>
                <FormButton onClick={this.handlePrintBloodCollectionButtonClick} style={{marginRight: '20%'}}>
                  Print out Blood Collection forms for your patient(s)
                </FormButton>
              </Col>
            </Row>

            <Row>
              <Col md={6}>
                <FormButton onClick={() => this.handleRedirect('genome')} style={{marginLeft: '20%', width: '78.26%'}}>
                  Back to Genome.One
                </FormButton>
              </Col>
              <Col md={6}>
                <FormButton onClick={() => this.handleRedirect('home')} style={{marginRight: '20%', width:'80%'}}>
                  Order new Test
                </FormButton>
              </Col>
            </Row>

        </ConfirmationBox>

        <div className="printMe">
          {(this.state.print === 1) && 
            <PrintRecord 
              orderTestModule={this.props.route.data.orderTestModule}
              patientDetails={this.props.route.data.patientDetailsModule}
              clinicalInfo={this.props.route.data.clinicalInfoModule}
              familyMember={this.props.route.data.familyMembersModule}
              clinicianDetails={this.props.route.data.clinicianDetailsModule}
              billingInfo={this.props.route.data.billingInfoModule}
            />
          }
          {(this.state.print === 2) &&
            <PrintBloodCollection 
              orderTestModule={this.props.route.data.orderTestModule}
              patientDetails={this.props.route.data.patientDetailsModule}
              clinicalInfo={this.props.route.data.clinicalInfoModule}
              familyMember={this.props.route.data.familyMembersModule}
              clinicianDetails={this.props.route.data.clinicianDetailsModule}
              billingInfo={this.props.route.data.billingInfoModule}
            />
          }
        </div>
      </div>
    );
  }
}

export default Confirmation;