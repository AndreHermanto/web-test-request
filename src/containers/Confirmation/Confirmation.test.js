import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestUtils from 'react-addons-test-utils';
import Confirmation from './index';

const patientData = {
  "lastName":"abcabc",
  "firstName":"Ngabcuyen",
  "dob":"1/February/1917",
  "medicalRecordNo":"123123",
  "gender":"Male",
  "genderOther":"",
  "ethnicity":"Asian",
  "deceased":false,
  "sampleSource":"",
  "consent":false
}

const familyData = {"familyMembers":[{"familyMemberDetails":{"lastName":"abc@abc.com","firstName":"abc@abc.com","dob":"2/January/1918","medicalRecordNo":"abc@abc.com","gender":"Female","genderOther":"","ethnicity":"","deceased":false,"sampleSource":"","consent":false},"familyMemberClinicalInfo":{"clinicalInfo":"abc@abc.com","affected":true,"relevantInvestigation":"asdasdasd","familyHistory":"asdasdasdasd","consangunity":false}},
  {"familyMemberDetails":{"lastName":"ab.com","firstName":"abcsadkfhkladsfj","dob":"2/January/1918","medicalRecordNo":"abc@abc.com","gender":"Male","genderOther":"","ethnicity":"","deceased":true,"sampleSource":"","consent":false},"familyMemberClinicalInfo":{"clinicalInfo":"abc@abc.com","affected":false,"relevantInvestigation":"asdasdasdasd","familyHistory":"asdasd","consangunity":true}}]};  
const clinicianDetails = {
  "providerNumber":"as","medicalSpecialty":"as","firstName":"John",
  "lastName":"Doe","organisation":"as","phone":"as","email":"asfd@abc.com","fax":"as",
  "copyToHCP":[{"firstName":"asdf","lastName":"asdf","organisation":"adsf","email":"asfd@abc.com"},
  {"firstName":"asdf","lastName":"asdf","organisation":"adsf","email":"asfd@abc.com"}],
  "copy":false
}

const clinicalData = {
  clinicalInfo: "Dissuade ecstatic and properly saw entirely sir why laughter endeavor. In on my jointure horrible margaret suitable he followed speedily. Indeed vanity excuse or mr lovers of on. By offer scale an stuff. Blush be sorry no sight. Sang lose of hour then he left find",
  consangunity: false,
  familyHistory: "kasdfbljkbfalhfasvfkjabskldfbkjsdfbkdashfvdjhfhb;kfjv;hfv;fjbafhb;akfd",
  relevantInvestigation:"asdkfhgasdlfhabsdlfjhalsfdjh"
}

const billingInfo = {
  payer:"abc abc",
  phone:"+61412334477",
  payerEmail:"abc@gmail.com",
  firstName:"abc",
  lastName:"abc",
  billOption:"Private"
}

const testData = {
  test: {
    id:'7052d137-a166-48b0-a52e-e05a167bd176',
    label: 'Consequatur adipisci modi laudantium tenetur ea exercitationem id',
    geneLists:[{
      type: "complete",
      genes: ["GZOW","KIHG","DRCO","DWBY","ZMZY","BMBP","OLZO","PZOA"]
    }]
  }
}

const confirmationData = {
  orderTestModule: testData,
  patientDetailsModule: patientData,
  clinicalInfoModule: clinicalData,
  familyMembersModule: familyData,
  clinicianDetailsModule:clinicianDetails,
  billingInfoModule: billingInfo
}

const props = {
  route: {
    onChange: jest.fn(),
    onClean: jest.fn(),
    data: confirmationData
  },
  router:['/step1']
};

describe('Confirmation test', () => {
  test('Confirmation page renders without crashing', () => {
    const page = renderer.create(React.createElement(Confirmation)).toJSON();
    expect(page).toMatchSnapshot();
  });
  
  test('handlePrintRecordButtonClick changes the print state to 1', () => {
    const page = TestUtils.renderIntoDocument(React.createElement(Confirmation, props));
    page.handlePrintRecordButtonClick();
    expect(page.state.print).toEqual(1);
    page.handlePrintButtonClick(2);
    expect(page.state.print).toEqual(2);
  });
  
  test('handlePrintBloodCollectionButtonClick changes the print state to 2', () => {
    const page = TestUtils.renderIntoDocument(React.createElement(Confirmation, props));
    page.handlePrintBloodCollectionButtonClick();
    expect(page.state.print).toEqual(2);
  });

  test('handleNext works', () => {
    var view = TestUtils.renderIntoDocument(React.createElement(Confirmation, props));                                         
    view.handleRedirect('home');
    expect(view.props.router.pop()).toEqual('/step1');
  });
});
