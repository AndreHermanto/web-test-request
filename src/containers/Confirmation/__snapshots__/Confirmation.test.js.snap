exports[`Confirmation test Confirmation page renders without crashing 1`] = `
<div>
  <div
    className="noPrint kEpFkI">
    <span
      className="idaOVj" />
    <span
      className="eTimuZ" />
    <div
      className="kZoPjc" />
    <h1
      className="eikxtc">
       Thank you!
    </h1>
    <div
      className="kZoPjc" />
    <h4
      className="bpcQEW">
      We\'ve sent you an email with all the details of your order!
    </h4>
    <h4
      className="bpcQEW">
      You can now:
    </h4>
    <div
      className="row">
      <div
        className="col-md-6">
        <button
          className="bBdbjl"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "20%",
            }
          }>
          Print out form for your records
        </button>
      </div>
      <div
        className="col-md-6">
        <button
          className="bBdbjl"
          onClick={[Function]}
          style={
            Object {
              "marginRight": "20%",
            }
          }>
          Print out Blood Collection forms for your patient(s)
        </button>
      </div>
    </div>
  </div>
  <div
    className="printMe" />
</div>
`;
